bot/cogs/utils.py

```diff
@@ -244,15 +245,20 @@ def get_pep_zero_embed() -> Embed:
 
         return pep_embed
 
-    @async_cache(arg_offset=1)
-    async def get_pep_embed(self, pep_nr: int, ctx: Context) -> Optional[Embed]:
+    @async_cache(arg_offset=2)
+    async def get_pep_embed(self, ctx: Context, pep_nr: int) -> Optional[Embed]:
         """Fetch, generate and return PEP embed."""
-        if pep_nr not in self.peps:
-            log.trace(f"PEP {pep_nr} was not found")
-            not_found = f"PEP {pep_nr} does not exist."
-            embed = Embed(title="PEP not found", description=not_found, colour=Colour.red())
-            await ctx.send(embed=embed)
-            return
+        while True:
+            if pep_nr not in self.peps and (self.last_refreshed_peps + timedelta(minutes=30)) > datetime.now():
+                log.trace(f"PEP {pep_nr} was not found")
+                not_found = f"PEP {pep_nr} does not exist."
+                embed = Embed(title="PEP not found", description=not_found, colour=Colour.red())
+                await ctx.send(embed=embed)
+                return
+            elif pep_nr not in self.peps:
+                await self.refresh_peps_urls()
+            else:
+                break
```

Comment:
It's confusing to follow the control flow of this loop. This could be simplified without a loop: 

```py
if pep_nr not in self.peps and (self.last_refreshed_peps + timedelta(minutes=30)) <= datetime.now():
    await self.refresh_peps_urls()

if pep_nr not in self.peps:
    ...
    await ctx.send(embed=embed)
    ...
```